crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -356,21 +356,19 @@ def upgrade_controller_nodes
         drbd_nodes = NodeObject.find("drbd_rsc:*")
         return upgrade_drbd_clusters unless drbd_nodes.empty?
 
-        founder = NodeObject.find(
-          "state:crowbar_upgrade AND pacemaker_founder:true"
-        ).first
+        founder = NodeObject.find("pacemaker_founder:true").first
         cluster_env = founder[:pacemaker][:config][:environment]
 
         non_founder = NodeObject.find(
-          "state:crowbar_upgrade AND pacemaker_founder:false AND " \
+          "pacemaker_founder:false AND " \
           "pacemaker_config_environment:#{cluster_env}"
         ).first
 
         return false unless upgrade_first_cluster_node founder.name, non_founder.name
```

Comment:
if the founder is in state ```ready``` here already, it indicates that everything including ```node_api.join_and_chef``` succeeded. Shouldn't we just skip those steps then in ```upgrade_first_cluster_node```?